src/webui/www/private/views/preferences.html

```diff
@@ -3172,16 +3180,19 @@
             settings["i2p_inbound_length"] = Number(document.getElementById("i2pInboundLength").value);
             settings["i2p_outbound_length"] = Number(document.getElementById("i2pOutboundLength").value);
 
-            // Send it to qBT
-            window.parent.qBittorrent.Cache.preferences.set(settings)
-                .then(() => {
-                    // Close window
-                    window.parent.location.reload();
-                    window.parent.qBittorrent.Client.closeWindow(document.getElementById("preferencesPage"));
-                }).catch((error) => {
-                    alert("QBT_TR(Unable to save program preferences, qBittorrent is probably unreachable.)QBT_TR[CONTEXT=HttpServer]");
-                    window.parent.qBittorrent.Client.closeWindow(document.getElementById("preferencesPage"));
-                });
+            // Send data to qBT
+            const results = await Promise.allSettled([
+                window.parent.qBittorrent.ClientData.set(clientData),
+                window.parent.qBittorrent.Cache.preferences.set(settings)
+            ]);
+            if (results.every((result) => result.status === "fulfilled")) {
+                window.parent.location.reload();
+                window.parent.qBittorrent.Client.closeWindow(document.getElementById("preferencesPage"));
+            }
+            else {
+                // keep window open so user can reattempt saving
+                alert("QBT_TR(Unable to save preferences, qBittorrent is probably unreachable.)QBT_TR[CONTEXT=HttpServer]");
```

Comment:
Could you not alternate the behavior in this unrelated PR??
```suggestion
                alert("QBT_TR(Unable to save preferences, qBittorrent is probably unreachable.)QBT_TR[CONTEXT=HttpServer]");
                window.parent.qBittorrent.Client.closeWindow(document.getElementById("preferencesPage"));
```

Reply:
Will move to a separate PR